share: Support choosing an account, if there are multiple

rajveermalviya opened the pull request:
| Light | Dark |
| - | - |
| <img width="1118" height="2384" alt="image" src="https://github.com/user-attachments/assets/df5114a8-87f6-4932-9730-c26f54633762" /> | <img width="1118" height="2384" alt="image" src="https://github.com/user-attachments/assets/5ae65586-5819-47cf-a588-95df7e7c610c" /> |

Tests are TODO and will happen as part of #1787 in a follow-up PR.

Fixes: #1779

chrisbobbe commented:
Thanks for building this!

I think my ideal version of this would have a slightly different UX:

- We don't add a new page, but instead show a modified version of the original page:
  - It has a nullable "currently selected account" state, controlled by a "Choose account" button somewhere like in the app bar
    - The button shows which account is selected, if any, making it persistently visible on the page (this should be helpful for accessibility; relying less on the user's memory/attention)
    - When pressed, the button opens a dialog with a list of accounts to select from
  - The page shows per-account data for the currently selected account (via `PerAccountStoreWidget`), or placeholder text if no account is selected.

I've pushed a rough draft of an approach like this, but it has some gaps:
- #1036
- Storing the realm logo in `PerAccountStore`
- Some UI strings to finalize
- maybe some visual design adjustments

If this approach sounds good, would you PTAL? Several of the commits are just cherry picks from #1802.

chrisbobbe commented:
Pushed with a rebase past #1802.

rajveermalviya commented:
Thanks @chrisbobbe for those draft commits.

I've sent a new revision as a separate PR (#1883) to preserve these draft commits. That PR addresses some of the points mentioned at https://github.com/zulip/zulip-flutter/pull/1816#issuecomment-3204157786. Modulo the nullable currently selected account state, instead it selects the current account as the initial account, PTAL.
